verticapy/machine_learning/metrics/classification.py

```diff
@@ -288,31 +308,58 @@ def critical_success_index(
     y_score: str
         Prediction.
     input_relation: SQLRelation
-        Relation to use for scoring. This relation can 
-        be a view, table, or a customized relation (if 
-        an alias is used at the end of the relation). 
+        Relation  to  use  for  scoring.  This  relation 
+        can  be a view, table, or a customized  relation 
+        (if an alias is used at the end of the relation). 
         For example: (SELECT ... FROM ...) x
+    average: str, optional
+        The method used to  compute the final score for
+        multiclass-classification.
+            micro    : positive  and   negative  values 
+                       globally.
+            macro    : average  of  the  score of  each 
+                       class.
+            weighted : weighted average of the score of 
+                       each class.
+            scores   : scores  for   all  the  classes.
+    labels: ArrayLike, optional
+        List   of   the  response  column   categories.
     pos_label: PythonScalar, optional
-        To  compute  the metric, one of  the  response 
-        column  classes must be the positive one.  The 
-        parameter 'pos_label' represents this class.
+        Label to use to identify the positive class. If 
+        pos_label is NULL then the global accuracy will 
+        be computed.
 
     Returns
     -------
     float
         score.
     """
-    tn, fn, fp, tp = _compute_tn_fn_fp_tp(y_true, y_score, input_relation, pos_label)
-    csi = tp / (tp + fn + fp) if (tp + fn + fp != 0) else 0
-    return csi
+    return _compute_final_score(
+        _accuracy_score,
+        y_true,
+        y_score,
+        input_relation,
+        average=average,
+        labels=labels,
+        pos_label=pos_label,
```

Comment:
Instead i propose we can use the following **locals() argument to make it even more cleaner:

def sub_fun(fun_call,one_param,two_param,three_param):
  if fun_call==1:
    return one_param+two_param+three_param

def main_fun(one_param,two_param,three_param):
  return sub_fun(1,**locals())

main_fun(1,2,3)

Reply:
will do it in the future. Thanks a lot for the tip.